### ISSUE TITLE
MISC: User confusion on arriving into Citizen OS platform via invite link / reluctance to register first

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1376]
@kevincrepin @ilmartyrk Something that's been a little hard to test (as the testers are all usually aware that they are testing the Citizen OS platform, and so aren't in the exact same mindset as they would be in reality), but something that has come through quite strongly for a couple of participants (one in a pre-launch round, and one in ideation Round 2), is this:

When invited via a URL or QR code, participants can be surprised and confused to suddenly find themselves on the Citizen OS platform (which they have no idea what it is/does), and being asked to register and account, before even knowing what they are signing up for. I'll include a few other related insights as well, as I think they'll be relevant to re-thinking it. 

**Here's the example from the recent testing Round:**

- In the scenario, the user is given a flyer by a staff member at the British library:

_Let’s re-imagine the British Library! We have funding to make significant changes to the library - let’s decide together as a community how to spend the money._
_Please follow this QR code to join the conversation:_ 
_https://test.app.citizenos.com/topics/join/gljY9sbyoyQf__ 

-  She follows the link, and the first thing she sees is this. At which point her thoughts are: "Citizen OS? Why am I here, I thought it was something to do with the British Library". "I'm not keen on signing up for things that I don't need to sign up for". 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/cbeb5b5e-d5fc-4ab6-a618-b702366ee156)

- She goes through to the register page and starts to register (though seems likely she wouldn't in reality - she imagines that the library staff were really persuasive, for the sake of the test.). And says "I'm not sure what I'm signing up for, or what I'm going to get from signing up. I'm not sure why I'm signing up". So she gives her "junk email address" (which presumably she doesn't really check/use).  

- She then has to verify her account (still not really knowing what for)

- The next thing she sees is this. For a second before this popped up there was also the 'Join topic' pop-up, but she didn't see it as this topic tour one overlays it.  
 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/514bcce0-11b1-4d21-bc6b-3b7ec89a39b3)

- She starts taking the tour (though says she's not sure if she usually would, or is only as she's being observed), and says things along the lines of: 

That she is still confused about where she is, and what she's doing there. During the tour, says she is "working backwards to what this platform is doing", to try to find out where she is and what she’s doing there. She says she is “Doing a lot of detective work”. She says she would have liked if she is new to the platform, on the first page "a summary of what is the platform, and why I'm even here".  
She was expecting a survey, so wonders why she is on Citizen OS. She sees it is much more engaged than a survey, and that she'll be able to see other people's responses. She likes that, but says it's much more work than she expected - she thought it was just a quick survey, but now gets the impression it is a whole long process - so she's signed up for more than she anticipated.

She didn't mention this, but I'm seeing here that this text is quite admin-focused "you can hold a vote". Whereas, actually from her perspective it is more: "there may be a vote". 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/09c922b5-8e6f-4f76-bbf7-6cc7b9bfb67a)

- After she's done the tour, she gets to this and finally feels more what was expecting: "Oh, there's the British Library!" She reads the topic text, which explains all about the process, how long it will last with several phases, etc. She says that this information would have been what she wanted to see at the beginning.  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/a6ecc42f-3c8e-4eb6-a681-1151f78e2fa8)

- She never got to see the 'Join topic' pop-up initially, as the tour popped up on top of it, and it was no longer there when she finished, but when she later came to see it again (due to the bug of it repeatedly appearing after following an invite link), she was a bit put off by the text "Are you sure you want to join this topic?". She said it seems very dramatic, and makes her think "why wouldn't I?". 
 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/736a6055-9e9b-45ee-9bfe-a467f33fb914)

**Example from one of Sara's pre-redesign launch tests:**  

_User expressed that they wouldn’t have registered without seeing the details of the discussion_

**Related examples from the Round 2 ideation admin tests:**

Not exactly the same, but shows a reluctance to register: there were 2 admins testing as a team, and one of them was very reluctant to make an account, even for the purposes of testing. They looked together through absolutely everything they could before making an account (which the other one did). And when it came to inviting her to her to the topic, she was still reluctant until I explained that for the purpose of the test we could just invite her (to test out the invite function), but she wouldn't have to actually make an account. 

**Related example from Round 1 admin test:** 

One of the Round 1 admins was also keen that participants could see some of the topic before they have to log in. Suggesting perhaps a compromise that they can view the contributions without logging in, but have to log in if they want to reply. 

A couple of the other admins in past tests have queried the necessity of the 'Join topic' or 'Join group' pop-up, suggesting it puts another barrier in the way for participants. They want access to the topic to be super fast and smooth, otherwise people will be less likely to engage. (And getting them actually onto a new platform, is a key barrier to participation - this has come up a lot in the testing).  

**Suggestions**

Needs some thinking through how to make it as streamlined as possible, but some ideas:

- Get rid of the 'Join topic' pop-up or replace it with something more helpful to orientate participants? 
- Allow participants to see the topic before they have to register? (Easier for public than private topic I guess, but even for private ones, they should see something that shows them what they're going in to). 
- Resolve the issue of the tour popping up over the top of the join topic pop-up.  
- Have some sort of explanation for new users right at the beginning that explains what the platform even is, and why they are there if they were expecting to be somewhere else? 
- Make the tour text more tailored to participants? Or ideally, different texts for admin or participants, but I'm not sure how technically possible this is..

**Comment by kevincrepin** [comment]
Thanks for the breakdown, very interesting to read. I understand the reluctance to register, I mean, people have so many accounts all over the place it is a bigger issue  in general. 

_Allow participants to see the topic before they have to register? (Easier for public than private topic I guess, but even for private ones, they should see something that shows them what they're going in to)._
Not an easy one... I mean when you create a private topic, you set it to private for a reason. But I also understand that when people are invited some indication of what this is would be very beneficial to understanding what they're getting into. Maybe the title could still be visible to invitees?

I think the first thing the invitee should see is actually immediately the "join topic" popup which has the topic's title visible. This should reassure them that they are in the right place. Now if they press join, there are 3 options:
- They have an account and are still logged in so they just join the topic – easy.
- They have an account but aren't logged in (anymore) 
- They don't have an account at all

I think for the 2 last options we should redirect to the register screen and maybe the register screen should include a bit more info on the platform than just the form (needs a design). If the user actually has an account for them it's easy to go to the login screen from here and it should be less of a barrier for them anyway because they already know the platform (at least a bit).

Then we can get to the walkthrough.

**Comment by BeccaMelhuish** [comment]
@kevincrepin Actually I just noticed that for groups the flow is currently as such that when a user follows a group invite link, this is what they get. So hopefully just following the flow for topics could be quite an easy fix for @ilmartyrk?  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/ddf2d88d-469a-48fb-b506-fa7871ff5c35)

Plus, this 'Join group' pop-up seems preferable, in that:

- It doesn't include the slightly off-putting "Are you sure you want to join this?" 
- It has more information about the group than the topic one does about the topic. 

So maybe we could follow a similar format for topics?

But some suggestions for possible improvements to these pop-ups as well:

- What do you think if we pull through the group or topic image as well (for those that have them)? 
- Could/should we also include the creator of the group or topic and their avatar? 
- For the group one, can we include the group description (see this [issue here](https://github.com/citizenos/citizenos-fe/issues/1097) on that subject), 
- For the topic one, can we include the topic intro? 
- What if we have a little 'i' box at the bottom with a "What is Citizen OS, and why have I been directed here?' text, which doesn't show the answer, but if you click 'Read more' or similar, then you can see a little text explaining something the essence of which is "Citizen OS is a free, non-profit participation platform for groups and communities to make decisions together. If you're here, it's because someone wants you to take part in a collective decision-making process on a matter relevant to you. Follow the prompts to get involved and have your say!" (EDIT: I see now from the screenshot there is something similar already, but not sure where the link goes to. Maybe a text that's readable in the pop-up itself is preferable though?)
- Maybe when we give admins the chance to make a link/QR code/send invites, we can also show them a preview of the pop-up, with a message along the lines of 'This is the first thing participants will see when they follow your link/QR/invite. For maximum engagement, make sure it's clear what it's all about!". (Off the top of my head, needs work). 

I realise this might be a lot to fit in one pop-up :D But if there's a way to make it work, I think it would be helpful to the users.

**Comment by BeccaMelhuish** [comment]
Made a spin-off issue for doing something similar with the emails: https://github.com/citizenos/citizenos-fe/issues/1540

**Comment by BeccaMelhuish** [comment]
Also added a separate issue for to discuss the idea of removing them entirely for public topics/groups - it feels like they might not be necessary in those cases: https://github.com/citizenos/citizenos-fe/issues/1551

**Comment by BeccaMelhuish** [comment]
Connected to [this issue](https://github.com/citizenos/citizenos-fe/issues/1097) too, but hopefully both can be fixed in one go :) (I.e. the pop-up improved for both topics and groups)

**Comment by kevincrepin** [comment]
Taking this issue with [this one](https://github.com/citizenos/citizenos-fe/issues/1097).

But a bit hard to follow. When I invite someone here the first thing they see is the create account screen. But if they've logged in then they see a very simple join group or topic popup, nothing like the one here above... So I'm getting a bit lost in these versions and iterations tbh 😨 

But I made some updates to the topic and group invitation popups to include clearer info:
![Property 1=Join Group Not Logged In](https://github.com/user-attachments/assets/0d12952f-6c05-434e-bfb0-e26e9babd37c)
![Property 1=Join Topic Not Logged in](https://github.com/user-attachments/assets/29873b4d-68c7-4bbc-9317-3bbad6d6da89)

I would definitely lead ppl to the website though, not add even more info to this popup about what Citizen OS is. There's already so much content in this now and I kinda feel that we're trying to explain too much to too many people too many times :) It's good to have ppl informed but it comes at the cost of keeping these content chunks focused. Now it's like "this is the info on the topic and who invited you and what you can do once you access the topic and oh you need to log in to see it and btw do you know what Citizen OS is?" :)

**Comment by BeccaMelhuish** [comment]
@kevincrepin These designs look great to me! And I agree info on what Citizen OS is might be overload indeed :) 

A couple of thoughts: 
- What should we include in the invitation details for when someone's been invited to a topic or group via an URL or QR code (which should be quite often) rather than via email? I guess only rights is relevant, but then to have a box just for that seems overkill. 
- I'm thinking that for the group one we could also have a 'What you can do once you join a group:' block, and within it could be: 'Access any topics that have been added to the group; Take part in the discussions and decision-making.' As this would give them an idea of what a 'group' is (for anyone who is coming to COS for the first time and this is the first thing they see, might help them feel more willing to join it).   

And copy-wise:
- 'Topic rights' should be 'group rights' in the first one :) 
- 'Participate in discussions, ideations and voting' could be 'Take part in the discussions and decision-making', as the topics will be quite variable in terms of what phases happen/are even mentioned in them (e.g. a topic starting with voting won't even have the discussions or idea gathering tabs visible).  
- Let's leave out the 'Request to become a co-editor' for now, though would be great to be able to do so (not currently possible)

And we should definitely make this the first thing invitees see when they follow a link, right? Rather than the log-in screen which they currently get (before knowing anything about what they're getting into).

**Comment by kevincrepin** [comment]
Yep, would be the first thing they see instead of the vague login screen.

I think we could still show the rights as well. No reason to show it for invite via mail but then not when it's invite via QR or link. I would say it we always show it or we never show it :)

I updated the copy and also added the "what can you do..." to group:

![Property 1=Join Group Not Logged In](https://github.com/user-attachments/assets/ebf1f840-ad37-49fe-8062-cc3255dcfe82)
![Property 1=Join Topic Not Logged in](https://github.com/user-attachments/assets/03a8eebd-13eb-460a-86e2-c4c9a66f44dd)

@ilmartyrk designs live here: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=248-35545&t=PbHNhKXrvHwK1YT3-1

**Comment by BeccaMelhuish** [comment]
Perfection thank you @kevincrepin! I'll move this to 'ready for dev' already - but one thing to check: 

**For when the invite is via URL**, do we still include the 'Invited by' and write 'Share link' (or similar?) where the name would be, or we only include the rights?  

![image](https://github.com/user-attachments/assets/0d0c98b8-9796-4fe5-9cb6-5515871f69d2)

**Comment by kevincrepin** [comment]
it can be there for information, but I don't think it adds all too much value. But doesn't hurt either.

**Comment by BeccaMelhuish** [comment]
Should be fixed together with #1701 and #1706

**Comment by BeccaMelhuish** [comment]
**Dev task summary:** 

#1700 will be fixed first, so they see the join pop-up before login. Then: 

- [ ] Combine the join pop-ups with the invitation pop-ups for topics and groups and make the headings 'Join topic' and 'Join group' respectively. 
- [ ]  Implement new design for **group** join pop-up from above [here](https://github.com/citizenos/citizenos-fe/issues/1376#issuecomment-2507262550)
- [ ]  Implement new design for **topic** join pop-up from above [here again](https://github.com/citizenos/citizenos-fe/issues/1376#issuecomment-2507262550)
- [ ] When the user has arrived via a join link remove 'Invitation sent to' line
- [ ] When the user has arrived via a join link remove the 'Invited by' field
- [ ] (As per [#1701](https://github.com/citizenos/citizenos-fe/issues/1701), also add the 'View group without joining' or 'View topic without joining' button for public groups/topics). 
- [ ] Remove the join pop-up that appears when a user click on 'Join topic' top right in public topics, and just add them instantly (so it now says 'Joined Topic')
- [ ] Make sure the same flow works as we currently have in place - so that once users have followed an invite or join link, then logged in or registered and verified, they are then taken to the correct topic or group.

**Comment by ilmartyrk** [comment]
@kevincrepin @BeccaMelhuish Too much info for me to process at this point, but basically we would unify the join and invite link views and implement the changes as suggested?

Difficult to estimate at this point so putting it in the ~16h ballpark, but might be less

**Comment by BeccaMelhuish** [comment]
@ilmartyrk Yes that's right as far as I'm concerned though this makes me realised that this only works for join pop-up when someone is coming via a join link, not when they're clicking on the join button shown below. 

But seeing as we already automatically add people to public topics when they engage, without any pop-up or confirmation, I think let's also make it so that when a user click the 'Join' button shown below we just add them, unless @kevincrepin objects :) 

![Image](https://github.com/user-attachments/assets/9c3f4ae0-c0e3-4f37-8bd4-f17b47703e9c)

But let's only do this for public topics for now, seeing as for the time being we still have the important "By joining this group, you agree to your email address being made visible to the group's admin(s)" message for public groups. 

Have added 'Remove the join pop-up that appears when a user click on 'Join topic' topic right in public topics, and just add them instantly' to the task list. 

And seeing as we'll be combining the invite and join pop-ups now, then will specify to change the title of each to 'Join topic' and 'Join group' as that works for both :) 

Also @ilmartyrk I've added this as a note to be sure we don't lose what we did there somehow in the process: "Make sure the same flow works as we currently have in place - so that once they have followed an invite or join link, then logged in or registered and verified, they are then taken to the correct topic or group."

**Comment by kevincrepin** [comment]
Yes, I agree we can do this without a popup for public topics, not public groups.

**Comment by BeccaMelhuish** [comment]
@DiverVM just thought I should specify:

This text here on the join topic pop-up should be the topic title and intro if there is an intro, if no intro then it can be the start of the topic text. And for groups it's simple - just the group description :) 
![Image](https://github.com/user-attachments/assets/a25b2ce0-d830-445c-98aa-225e42e37978)

**Comment by DiverVM** [comment]
@BeccaMelhuish same as for topic I will make join group here without modal

<img width="1097" alt="Image" src="https://github.com/user-attachments/assets/b677cafa-433b-4973-81de-892931053dbb" />

**Comment by BeccaMelhuish** [comment]
@DiverVM Actually yes you can now, as we no longer show group members email addresses, no longer necessary to have the pop-up :) 

Which reminds me - I will make a separate issue if preferred, but we can now remove the text "By joining this group you agree to your email being visible to the group's admin) from the 'Join group' pop-up :)

**Comment by DiverVM** [comment]
no separate issue needed for this specific case

**Comment by ssin1901** [comment]
@BeccaMelhuish little confused here. To confirm, here is what I saw in the development environment

1. Logged in user clicks Join Topic and is Joined immediately, no popup (screen recording 1)
2. Logged out user doesn't see Join Button at all (screenshot1)

https://github.com/user-attachments/assets/d2101d89-510b-4791-b6ae-3c032cc74b1f

![Image](https://github.com/user-attachments/assets/0ff1397a-bd49-4948-938a-d3b2182838e2)

**Comment by BeccaMelhuish** [comment]
@ssin1901 1 is correct, but 2 isn't - there should still be the button, but then they get prompted to log in :) 

Though in this case I think make a separate issue for that one and we can focus on the new pop-up content/designs for this issue.

**Comment by BeccaMelhuish** [comment]
@ssin1901 1 is correct, but 2 isn't - there should still be the button, but then they get prompted to log in :) 

And here is the dev task summary if that helps clarify everything else to test here, there's quite a lot in one!  https://github.com/citizenos/citizenos-fe/issues/1376#issuecomment-2587737200

**Comment by ssin1901** [comment]
@BeccaMelhuish made a separate issue #1894. Can we move this to Pre-live because most of the issues are fixed?

**Comment by BeccaMelhuish** [comment]
Moved it back while we're still QA testing it :)

**Comment by BeccaMelhuish** [comment]
Will move this to ready for pre-live now that we have sub-issues for it :)
